### ISSUE TITLE
Troubles understanding how to limit search access of user in ldapsearch group.

### ISSUE THREAD

**Issue by Buco7854** [goauthentik/authentik#10403]
**Question:**

Hello,

I have some questions regarding LDAP in Authentik. In many self-hosted apps, when a user needs to log in using LDAP, the app performs a User Bind request to authenticate. The issue is that if the user is not in the ldapsearch group, the authentication fails.

I could add all users to the ldapsearch group, but I have some security concerns. If I understand correctly, any user in this group will be able to perform search queries. This means that any user could potentially access information via LDAP that they aren't supposed to have, right?

If so, how can I mitigate this?

**Example:**

For example in calibre web, when trying to login, the calibre service account with ldap search right will look for the user, and once it has found it, it tries to bind with this user (that has not ldapsearch since only service account has) and so it fails (if i understood correctly).

Here are the logs for this example:
```
2024-07-07T23:39:46.080802171Z INF bindDN=cn=service_account,ou=users,dc=example,dc=com client=192.168.1.66 event=User has access requestId=200348ed-a668-43b4-927a-cb03dd9e5544 timestamp=2024-07-07T23:39:46Z
2024-07-07T23:39:46.136793631Z INF bindDN=cn=service_account,ou=users,dc=example,dc=com client=192.168.1.66 event=Allowed access to search group=ldap_search requestId=200348ed-a668-43b4-927a-cb03dd9e5544 timestamp=2024-07-07T23:39:46Z
2024-07-07T23:39:46.136838699Z INF bindDN=cn=service_account,ou=users,dc=example,dc=com client=192.168.1.66 event=Bind request requestId=200348ed-a668-43b4-927a-cb03dd9e5544 timestamp=2024-07-07T23:39:46Z took-ms=1183
2024-07-07T23:39:46.138643858Z INF attributes=[] baseDN=dc=example,dc=com bindDN=cn=service_account,ou=users,dc=example,dc=com client=192.168.1.66 event=Search request filter=(&(cn=user)) requestId=3c339a1a-d4d4-43bd-9308-b5f262588804 scope=Whole Subtree timestamp=2024-07-07T23:39:46Z took-ms=0
2024-07-07T23:39:47.445587632Z INF bindDN=cn=service_account,ou=users,dc=example,dc=com client=192.168.1.66 event=User has access requestId=41112b50-6950-4dd0-a3dd-7964fd80d72e timestamp=2024-07-07T23:39:47Z
2024-07-07T23:39:47.495344947Z INF bindDN=cn=service_account,ou=users,dc=example,dc=com client=192.168.1.66 event=Allowed access to search group=ldap_search requestId=41112b50-6950-4dd0-a3dd-7964fd80d72e timestamp=2024-07-07T23:39:47Z
2024-07-07T23:39:47.495380172Z INF bindDN=cn=service_account,ou=users,dc=example,dc=com client=192.168.1.66 event=Bind request requestId=41112b50-6950-4dd0-a3dd-7964fd80d72e timestamp=2024-07-07T23:39:47Z took-ms=1352
2024-07-07T23:39:47.496714781Z INF attributes=[] baseDN=dc=example,dc=com bindDN=cn=service_account,ou=users,dc=example,dc=com client=192.168.1.66 event=Search request filter=(&(cn=user)) requestId=bbca1a7f-6598-4736-80f3-ef938f1e881c scope=Whole Subtree timestamp=2024-07-07T23:39:47Z took-ms=0
2024-07-07T23:39:47.501236556Z {"bindDN":"uid=user,ou=people,dc=example,dc=com","client":"192.168.1.66","event":"No provider found for request","level":"warning","request":"bind","requestId":"8f1fe7ef-758a-4024-985e-23030c544399","timestamp":"2024-07-07T23:39:47Z"}
2024-07-07T23:39:47.501261592Z INF bindDN=uid=user,ou=people,dc=example,dc=com client=192.168.1.66 event=Bind request requestId=8f1fe7ef-758a-4024-985e-23030c544399 timestamp=2024-07-07T23:39:47Z took-ms=0
```

**Comment by Buco7854** [comment]
Ok nevermind i just got it.

**Comment by aep** [comment]
and what was the solution?

**Comment by Buco7854** [comment]
Actually it wasn't really related to authentik

The permission thing works as other apps do it's just that LDAP search allow for search.

I couldn't bind because I actually setup LDAP search as the only group to access LDAP at all.

Apart of that I had some issue with a config and some user attributes that where imported from an old LDAP source.

The was no cn in distinguishedName attribute so binding also failed cause of that.

Anyway the issue got solved when I checked "Is open LDAP server" in calibre config so this attribute was not used anymore and I just removed it.

That's all, what happen is that with those 3 things it kind of confused me while debugging so figuring out those 3 took some time.

Ultimately Authentik works as expected.
